### ISSUE TITLE
Doesn't work with ps3 remote accelerometer

### ISSUE THREAD

**Issue by mak448a** [SagaPDev/Godot-SDL-Gamepad-Test#15]
Title. It doesn't work at all. I did press "C" to calibrate at the beginning.

**Comment by mak448a** [comment]
AntiMicroX (a controller mapping program) detects it properly, under accelerometer. Is there a way to implement this into this project?
![image](https://github.com/user-attachments/assets/108b1190-d7b6-400f-9496-21dc618abc56)

**Comment by SagaPDev** [comment]
i was interested in looking into this but because the dualshock 3 only has accelerometers it would be a limited implementation, first i need to be able to read the accelerometer sensor state from SDL, currently is not being detected

**Comment by mak448a** [comment]
It seems to be able to do tilt tracking somewhat accurately (only on a few axes), so is there an accelerometer?


https://github.com/user-attachments/assets/47d6f0dd-fece-4a99-91e7-f93628cfb00d

**Comment by SagaPDev** [comment]
yes there is, but they are not being detected on my end, let my try on another platform, that is windows you're in right?
Edit:
In windows it gamepad does not work, and on Linux the gamepad works but sensor is not being detected, how are you getting the sensor to show up? (i'm also using antimicrox)

**Comment by mak448a** [comment]
I'm on Fedora Linux 40 KDE Spin. I'm using AntiMicroX as an AppImage. I was using it through bluetooth, but it also works with a cable. [This](https://wiki.archlinux.org/title/Gamepad#PlayStation_3_controller) might help?

**Comment by SagaPDev** [comment]
The controller works and the sensor is detected in /dev/event
Works fine in RPCS3 via evdev and SDL

https://github.com/user-attachments/assets/f7561ca0-4e7d-4df0-92d4-e01ee7e809af


But it does not respond in Antimicrox
![image](https://github.com/user-attachments/assets/39c3ea02-4c22-4af5-bf2f-c0fc46918e6e)
Or when i try to get sensor data on my own [code](https://github.com/SagaPDev/sdl-gyro-test/commits/master/)   
![image](https://github.com/user-attachments/assets/fc84d202-e99e-4605-8546-b85b86d2f371)

here is a precompiled binary of that project if you want to try: [sdl_test.zip](https://github.com/user-attachments/files/17311511/sdl_test.zip)
don't know what it could be, tested on fedora 40 too

Be aware that even if it works because it is only relaying on accelerometers,  the functionality will be limited, no precise orientation, or accurate gyro aim for example

**Comment by mak448a** [comment]
Did you go to the accelerometer section in AntiMicroX? Your screenshot shows the stick calibration. Also, I'll try your precompiled binary.

https://github.com/user-attachments/assets/2c24c927-7f7b-4372-a765-aa4d2e4465d1

**Comment by mak448a** [comment]
My results with your project are the same. It says no accelerometer too.

**Comment by mak448a** [comment]
My user is in the groups `input uinput` if that helps

**Comment by SagaPDev** [comment]
I'm on the input group seems like no "uinput" on my system, but does not seem to be a problem with permissions, i can read the sensor status and some apps can access the sensor no problem

RPCS3-------->SDL✓    evdev✓
Dolphin------>SDL✗    evdev✓
Antimicrox-> SDL✗


I also tested on a clean fedora install

Are you sure you didn't do any extra steps to make the sensor appear on antimicrox, or my SDL test aplication?
If it not much to ask do you have access to a clean install?

My device is (taken from lsusb):
`ID 054c:0268 Sony Corp. Batoh Device / PlayStation 3 Controller`
evtest utility they appears as:
```
Sony PLAYSTATION(R)3 Controller Motion Sensors
Sony PLAYSTATION(R)3 Controller
```
the SDL api call it "PS3 Controller" with GUID:0300afd34c05000068020000118100001356616

**Comment by mak448a** [comment]
The only thing from stock that I can think of is the groups and the udev rules that antimicrox asks for. I don't have a clean installation lying around anywhere though. I could probably try a live usb when I have a lot of time, but I probably won't get around to that.
Also, I'm using it over bluetooth if that helps.

**Comment by SagaPDev** [comment]
No problem, thanks for the report.
It does work on you setup so instead of finding out why the sensor does not work, it would be better to just disable the gyro on my switch-pro and see if the gamepad-motion-helper library works with just that, and do the necessary changes in the add-on

i will report back if i make progress
